### PR TITLE
Modernization

### PR DESCRIPTION
**Pull request by maxsbelt**
# API
Package API remains the same. Package exports two methods: `createTypes` and `composeReducer`. 

## createTypes
- method consumes the same arguments: `namespace` and `types`.
- added method arguments validation with user friendly error message:
screen shot 2018-08-01 at 22 54 43

- as user I can make a typo in action creator. Currently you will see error about undefined action type. It helps but we can make it better. With power of `Proxy` and `Error.captureStackTrace` apis we can show exact line of code where typo was done. Cool, right?

Before:

screen shot 2018-08-01 at 22 48 58

After:

screen shot 2018-08-02 at 22 54 54

## composeReducer
- current arguments still supported but prints deprecation warning:

screen shot 2018-08-02 at 22 57 37

- list of arguments was replaced with single object argument. Added user friendly error message if user provides invalid object argument:

screen shot 2018-08-02 at 23 01 39

- Almost all object argument parameters are just named parameters of previous API, but there is new one `types` parameter. Initially we provided `namespace` parameter for `action type -> reducer` mapping creation. But we can go further. Instead of passing `namespace` parameter we can pass our action types (created with `createTypes` method). In this case we'll already have necessary mapping and, moreover, we can validate provided reducers by searching corresponded action:

screen shot 2018-08-02 at 23 11 08

# Linting and formatting
I decided to use `prettier` for code formatting purposes. It's a good option for such small codebases. For general linting purposes used `eslint` config. It provides errors highlighting in text editors:

screen shot 2018-08-02 at 23 06 00

# Testing
`mocha` was replaced with faster, current industry standard `jest` package. New tests provides 100% code coverage:

screen shot 2018-08-01 at 22 24 57

# CI
Added `.travis.yml` config for both LTS (`8.11.3`) and latest(`10.7.0`) node versions. CI validates both code formatting and tests. We just need to enable `travis CI` for this repository.

screen shot 2018-08-01 at 21 24 25

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  return error
+}
+
+const TYPES_PROXY_HANDLER = {
+  get(target, code) {
+    if (code in target) {
+      return target[code]
+    }
+
+    const keys = Object.keys(target).join(', ')
```
Comment (kimrgrey) [inline src/utils.js]: Looks like this variable is not used. Am I missing something?

File: package.json [hunk 2]
```diff
@@ -2,10 +2,28 @@
   "name": "redux-compose-reducer",
   "version": "1.2.0",
   "description": "Reducer's composer for Redux",
-  "main": "index.js",
+  "main": "lib/index.js",
+  "unpkg": "dist/redux.js",
```
Comment (kimrgrey) [inline package.json]: Looks like copy / paste mistake. Should it be `dist/index.js`?

**Review by kimrgrey — COMMENTED** [review]
I would also add few words about new capabilities into `README`. But it can be done as separated issue.

File: src/utils.js (hunk 1 above)
Comment (maxsbelt) [reply src/utils.js]: https://github.com/kimrgrey/redux-compose-reducer/pull/3/commits/e6435005a1bc0034cb59079e52d7efff4fa678e4

File: package.json (hunk 2 above)
Comment (maxsbelt) [reply package.json]: https://github.com/kimrgrey/redux-compose-reducer/pull/3/commits/bd626cdafe82fede73f1ac648a1a9b6615166efd

**Review by kimrgrey — APPROVED** [review]

**Pull request merged**